tiny_dnn/layers/layer.h

```diff
@@ -299,6 +299,9 @@ class layer : public node {
    *important
    **/
   virtual serial_size_t fan_in_size() const { return in_shape()[0].width_; }
+  virtual serial_size_t fan_in_size(serial_size_t) const {
+    return fan_in_size();
+  }
```

Comment:
This is to allow initializing weights of different sizes inside a layer, needed for recurrents.